pkg/components/mysql/mysql.go

```diff
@@ -56,6 +55,8 @@ func Cluster(mattermost *mattermostv1alpha1.ClusterInstallation) *mysqlOperator.
 	if mattermost.Spec.Database.InitBucketURL != "" && mattermost.Spec.Database.BackupRestoreSecretName != "" {
 		mysql.Spec.InitBucketURL = mattermost.Spec.Database.InitBucketURL
 		mysql.Spec.InitBucketSecretName = mattermost.Spec.Database.BackupRestoreSecretName
+	} else if mattermost.Spec.Database.ExistingSecret != "" {
```

Comment:
I think this should be a separate `If` statement. In other words, we could use an existing secret as well as define backups.

Is that correct @cpanato?

Reply:
It makes sense, separated the check now :)